rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -171,17 +173,22 @@ abstract class ItemNode extends Locatable {
   }
 
   /** Gets a successor named `name` of this item, if any. */
-  pragma[nomagic]
+  cached
```

Comment:
Did something change in this PR that means `getASuccessor` should now be cached? Or would it already have made sense to cache it prior to this PR?

Reply:
I think it already made sense prior to this PR since the type inference stage relies on the predicate.